### PR TITLE
[PL-40190] Add API rate limit topic

### PR DESCRIPTION
**Pull request by brian-f-harness**
Add API rate limit topic

For reviewers, [a preview](https://64b9af0bad1ebc065dfea7a4--harness-developer.netlify.app/docs/platform/Resource-Development/APIs/rate-limits) is available.

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by pratmit — APPROVED** [review]

File: docs/platform/Resource-Development/16_APIs/rate-limits.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+title: Rate limits
+description: Learn about rate limits that help maximize the stability of the Harness API.
+sidebar_position: 4
+---
+
+The Harness API uses several mechanisms, including rate limits, to safeguard against bursts of incoming traffic and help maximize stability.
+
+Harness has several rate limiters in the API. Users who send multiple requests in quick succession may receive status code 429 error responses.
+
+A rate limiter restricts the number of requests received by the API within any given minute, after which Harness throttles requests from the IP address.
```
Comment (aimurphy) [inline docs/platform/Resource-Development/16_APIs/rate-limits.md:11]:
```suggestion
A rate limiter restricts the number of requests received by the API within any given minute, after which Harness throttles requests from the IP address.

Harness has several rate limiters in the API. Users who send multiple requests in quick succession may receive status code 429 error responses.
```

It makes more sense to me if rate limiter is defined before you explain what happens when it's triggered.

File: docs/platform/Resource-Development/16_APIs/rate-limits.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+:::info note
+API throttling is a type of rate limiting used to control the amount of traffic that an API can handle. It is a way of limiting the number of requests the API accepts in a given period.
+:::
+
+Harness API rate limits include:
+
+- Any call: Harness allows 5000 requests every 10 seconds (30,000 requests per minute) per IP address.
+- API calls: Harness allows 1000 requests per API key per minute.
+- GraphQL APIs: Harness allows up to 1000 requests a minute per IP address.
```
Comment (aimurphy) [inline docs/platform/Resource-Development/16_APIs/rate-limits.md:21]:
```suggestion
- GraphQL APIs: Harness allows up to 1000 requests per minute per IP address.
```

File: docs/platform/Resource-Development/16_APIs/rate-limits.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- Large requests (character size > 500,000): 1 payload every 10 seconds.
+- Reset password: 5 calls per minute per IP address.
+- User invites: 10 calls per minute for IP address.
```
Comment (aimurphy) [inline docs/platform/Resource-Development/16_APIs/rate-limits.md:24]:
```suggestion
- User invites: 10 calls per minute per IP address.
```

**Review by aimurphy — APPROVED** [review]
approved with comments

**Pull request merged**